src/cdomains/apron/linearTwoVarEqualityDomain.apron.ml

```diff
@@ -0,0 +1,770 @@
+    let multiply_with_Z number coeff_var_list =
+      List.map (fun (coeff, var) -> (Z.(number * coeff, var))) coeff_var_list in
+    let multiply a b =
+      (* if one of them is a constant, then multiply. Otherwise, the expression is not linear*)
+      match a, b with
+      | [(a_coeff, None)], b -> multiply_with_Z a_coeff b
+      | a, [(b_coeff, None)] -> multiply_with_Z b_coeff a
+      | _ -> raise NotLinearExpr
+    in
+    let mpqf_to_Z x =
+      if not(Z.equal (Mpqf.get_den x) Z.one) then raise NotIntegerOffset
+      else Mpqf.get_num x in
+    let rec convert_texpr texp =
+      begin match texp with
+        (*If x is a constant, replace it with its const. val. immediately*)
+        | Cst x ->
+          (let open Coeff in
+           match x with
+           | Interval _ -> failwith "Not a constant"
+           | Scalar (Float x) -> raise NotIntegerOffset
+           | Scalar (Mpqf x) -> [(mpqf_to_Z x, None)]
+           | Scalar (Mpfrf x) -> raise NotIntegerOffset)
+        | Var x ->
+          let var_dim = Environment.dim_of_var t.env x in
+          begin match t.d with
+            | None -> [(Z.one, Some var_dim)]
+            | Some d ->
+              (if Option.is_some (fst d.(var_dim)) then [(Z.one, fst d.(var_dim))]
+               else [])
+              @ [(snd d.(var_dim), None)]
+          end
+        | Unop (u, e, _, _) ->
+          begin match u with
+            | Neg -> negate (convert_texpr e)
+            | Cast -> convert_texpr e (*Ignore since casts in apron are used for floating point nums and rounding in contrast to CIL casts*)
+            | Sqrt -> raise NotLinearExpr end
+        | Binop (b, e1, e2, _, _) ->
+          begin match b with
+            | Add -> List.concat [convert_texpr e1; convert_texpr e2]
+            | Sub -> List.concat [convert_texpr e1; negate (convert_texpr e2)]
+            | Mul -> multiply (convert_texpr e1) (convert_texpr e2)
+            | _ -> raise NotLinearExpr end
+      end
+    in match convert_texpr texp with
+    | exception NotLinearExpr -> None
+    | x -> Some(x)
+
+  let get_coeff (t: t) texp =
+    let d = Option.get t.d in
+    let expr = Array.init (Environment.size t.env) (fun _ -> Z.zero) in
+    let constant = ref (Z.zero) in
+    match get_coeff_vec t texp with
+    | None -> None (*The (in-) equality is not linear, therefore we don't know anything about it. *)
+    | Some cv's ->
+      let update (c, v) =
+        match v with
+        | None -> constant := Z.(!constant + c)
+        | Some idx -> match d.(idx) with
+          | (Some idx_i, c_i) -> constant := Z.(!constant + (c * c_i));
+            expr.(idx_i) <- Z.(expr.(idx_i) + c)
+          | (None, c_i) -> constant := Z.(!constant + (c * c_i))
+      in
+      List.iter update cv's;
+      let var_count = GobArray.count_matchingi (fun _ a -> a <> Z.zero) expr in
+      if var_count == 0 then Some (None, !constant)
```

Comment:
Also, please be careful with comparing values of type `Z.t`. Neither `=` nor `==` can be used here.

Reply:
I changed the comparision in line 218 but `var_count` is of type `int` therefore its comparision should be correct